File: RQ1/CodeForces-Automatic/122A.py
def main():
    # PRE:
    # Input Format:
    # The single line contains an integer n (1 ≤ n ≤ 1000) — the number that needs to be checked.
    # POST:
    # Program Description:
    # Petya loves lucky numbers. Everybody knows that lucky numbers are positive integers whose decimal representation contains only the lucky digits 4 and 7. For example, numbers 47, 744, 4 are lucky and 5, 17, 467 are not.Petya calls a number almost lucky if it could be evenly divided by some lucky number. Help him find out if the given number n is almost lucky.
    # Output Format:
    # In the only line print "YES" (without the quotes), if number n is almost lucky. Otherwise, print "NO" (without the quotes).
    s = input()
    n = int(s)
    
    if n % 4 ==0 or n % 7 == 0 or n % 47 == 0:
          print('YES')
    else:
          s = s.replace('4', '')
          s = s.replace('7', '')
          if s == '':
                print('YES')
          else:
                print('NO')
    
    
    
    
    
    

File: RQ1/CodeForces-Automatic/122A.c
#include <stdio.h>
#include <assert.h>

int is_lucky(int num) {
    while (num > 0) {
        int r = num % 10;
        if (r != 4 && r != 7) {
            return 0;
        }
        num = num / 10;
    }
    return 1;
}

int main() {
    int n;
    scanf("%d", &n);
    assert(n >= 1 && n <= 1000); // PRE: n is within the range 1 to 1000

    for (int num = 1; num <= 1000; ++num) {
        if (is_lucky(num) && n % num == 0) {
            printf("YES\n");
            assert(n % num == 0); // POST: n is almost lucky since n is divisible by lucky number
            return 0;
        }
    }
    printf("NO\n");
    assert(n % 1 != 0); // POST: If "NO" printed, n is not almost lucky
    return 0;
}